AIPscan/Reporter/templates/reports.html

```diff
@@ -405,44 +416,52 @@
   $("#largestAIPs").on("click", function() {
     var startdate = $('#startdate').val();
     var enddate = $('#enddate').val();
+    var limit = $('#largestAipsLimit').val();
+
+    if (!$.isNumeric(limit) || limit < 1) {
+      alert(LIMIT_ALERT);
+
+      return false;
+    }
+
     if (enddate < startdate) {
       alert(DATE_ALERT_START);
     } else {
-      var url = (
-        window.location.origin +
-        '/reporter/largest_aips/' +
-        '?amss_id=' +
-        storageServiceId +
-        '&start_date=' +
-        startdate +
-        '&end_date=' +
-        enddate +
-        '&storage_location=' +
-        storageLocationId +
-        '&limit=20'
-      );
+      var params = {
+        "amss_id": storageServiceId,
+        "start_date": startdate,
+        "end_date": enddate,
+        "storage_location": storageLocationId,
+        "limit": limit
+      }
+
+      var url = window.location.origin + "/reporter/largest_aips/?" + $.param(params);
       window.open(url);
     }
   });
   $("#largestFiles").on("click", function() {
     var startdate = $('#startdate').val();
     var enddate = $('#enddate').val();
+    var limit = $('#largestFilesLimit').val();
+
+    if (!$.isNumeric(limit) || limit < 1) {
+      alert(LIMIT_ALERT);
+
+      return false;
+    }
+
     if (enddate < startdate) {
       alert(DATE_ALERT_START);
     } else {
-      var url = (
-        window.location.origin +
-        '/reporter/largest_files/' +
-        '?amss_id=' +
-        storageServiceId +
-        '&start_date=' +
-        startdate +
-        '&end_date=' +
-        enddate +
-        '&storage_location=' +
-        storageLocationId +
-        '&limit=20'
-      );
+      var params = {
+        "amss_id": storageServiceId,
+        "start_date": startdate,
+        "end_date": enddate,
+        "storage_location": storageLocationId,
+        "limit": limit
+      }
+
+      var url = window.location.origin + "/reporter/largest_files/?" + $.param(params);
       window.open(url);
```

Comment:
You could try to use the URL API, it's [widely supported](https://caniuse.com/?search=URL) these days unless you still care about IE 11. E.g.:

```js
const url = new URL("/reporter/largest_files/", window.location.origin);
url.search = new URLSearchParams(params).toString();
window.open(url.href);
```

Reply:
Nice tip! Thanks!